mnt/reform2-nitrogen8m/kernel.nix

```diff
@@ -0,0 +1,92 @@
+
+let
+  systemImageSrc = fetchgit {
+    url = "https://source.mntmn.com/MNT/reform-system-image.git";
+    rev = "f34c9d93cef73c50979c1f45c68b8de961faceb5";
+    sha256 = "0a1gjnvb078q7f76rppidxc1sb4hb33wbxpzzs85il8i9n2nc56x";
+  } + "/reform2-imx8mq";
+in lib.overrideDerivation (buildLinux (args // {
+  inherit (linux_5_7) src version;
+
+  features = {
+    efiBootStub = false;
+    iwlwifi = false;
+  } // (args.features or { });
+
+  kernelPatches = let
+    patchDir = "${systemImageSrc}/template-kernel/patches";
+    reformPatches = map ({ name, extraConfig ? "" }: {
+      inherit name extraConfig;
+      patch = "${patchDir}/${name}.patch";
+    }) [
+      {
+        name = "0001-drm-bridge-Add-NWL-MIPI-DSI-host-controller-support";
+        extraConfig = ''
+          DRM_NWL_MIPI_DSI m
+        '';
+      }
+      { name = "0002-dt-bindings-display-bridge-Add-binding-for-NWL-mipi-"; }
+      {
+        name =
+          "0003-DCSS-v4-Add-support-for-iMX8MQ-Display-Controller-Subsystem";
+        extraConfig = ''
+          DRM_IMX_DCSS m
+        '';
+      }
+      { name = "4101-media-vb2-wait-for-dmabuf-fences"; }
+      { name = "4201-HACK-media-vb2-don-t-validate-buffer-length"; }
+      { name = "44f0bbdcf0433052b4e85940cb41d04c13fdad57"; }
+      { name = "git.linuxtv.org-28a202c55963386b8bc45bcc52029362e9aa0d33"; }
+      { name = "git.linuxtv.org-88d06362d1d052e4c844ac95a2ca308ed4d90452"; }
+      { name = "mnt1000-pcie-reparent-clocks"; }
+      { name = "mnt1001-pcie-support-internal-refclk-aspm"; }
+      { name = "mnt2000-audio-wm8960-add-dacslope-setting"; }
+      { name = "mnt2001-audio-sai-workaround-rate-matching"; }
+      { name = "mnt3000-imx-dcss-tweak-DCSS-pixel-rate-to-prevent-sporadic-d"; }
+      { name = "mnt3001-nwl-dsi-disable-bridge_mode_fixup-that-breaks-hs-vs-"; }
+      { name = "mnt3002-MNT-Reform2-add-simple-panel-Innolux-N125HCE-GN1"; }
+      {
+        name = "mnt3003-MNT-Reform-import-cadence-HDMI-driver-for-imx8mq-fro";
+        extraConfig = ''
+          DRM_CDNS_AUDIO m
+          DRM_CDNS_DP m
+          DRM_CDNS_HDMI m
+          DRM_CDNS_HDMI_CEC m
+          DRM_CDNS_MHDP m
+        '';
+      }
+      { name = "mnt3004-MNT-Reform-imx8mq-add-PHY_27M-clock"; }
+      { name = "mnt3005-MNT-Reform-imx8mq-DCSS-add-module-option-to-toggle-h"; }
+      { name = "mnt3006-MNT-Reform-imx8mq-add-PHY_27M-clock-missing-define"; }
+      {
+        name = "mnt3007-MNT-Reform-imx8mq-missing-kconf-makefile-for-cadence";
+        extraConfig = ''
+          DRM_IMX_CDNS_MHDP m
+          DRM_IMX_DCSS m
+        '';
+      }
+      {
+        name = "mnt3008-MNT-Reform-imx8mq-missing-makefile-change-for-imx-dc";
+        extraConfig = ''
+          DRM_IMX_DCSS m
+        '';
+      }
+    ];
+  in lib.lists.unique (kernelPatches ++ reformPatches ++ [{
+    name = "MNT-Reform-imx8mq-config";
+    patch = null;
+    extraConfig = builtins.readFile ./kernel-config;
+  }]);
```

Comment:
@samueldr what should we do with the kernel?

Reply:
This is still an open question about *everything* kernel related and "those kind" of platforms, where patches or outright complete kernel trees are needed.

I think we need to "sit down" and discuss that with the relevant key people: can we provide kernel builds in a useful manner, without making them part of Nixpkgs outright?

I'm thinking about a strategy similar to Mobile NixOS' own for reducing churn in Hydra, instead of tracking Nixpkgs, track channel updates. It means that it lags behind slightly, but doesn't needlessly queue builds... (Though that last part is not really an issue for kernels, few changes trigger rebuilds.)

I guess Flakes would help in this current situation, but adding a strict dependency on flakes is not great either.

Though one of the main important things about that is that managing non-mainline kernels through Nixpkgs is bound to become untenable. I already want to move the Raspberry Pi foundation kernels outside whenever possible, in addition to the few misc. probably-unused kernels still present.

Ugh, sorry to drop all that in a PR review request :/.

Reply:
Oh, additionally `linux_5_7` is bound to disappear. It's already marked EOL at kernel.org.

Reply:
I actually would not those packages to hydra and the official binary cache but have github repositories in nix-community. We have a diverse community and only a few people with access to the nixos core infrastructure. In nix community there is a 1TB cachix cache, which would allow to upload custom kernel builds and the whole process is a lot less bureaucratic.

Reply:
> Ugh, sorry to drop all that in a PR review request :/.

Not a problem, the hardware hasn't shipped en masse, so we have time to do this right.

Reply:
There is a level of trust around "just using a cachix repo", that I highly dislike. That's the main reason I don't publish anything for the Pinebook Pro WIP stuff, I don't want to have people either not realise they rely on an implicit trust on me, or have them trust what I do.

If the Nix community builder is transparent enough for that, it'd help, we'd probably be able to handle that for all NixOS on ARM things. I really want to move all the things in a centralized location with builds. @Mic92 care to open an issue and start the discussion about that at the right place (and tagging me in)?

Reply:
@samueldr I just realized that we don't have aarc64 builds in nix-community. Apart from that adding declarative jobs is supported https://github.com/nix-community/infra/blob/0374f082455b9cce0a369bdc53cc4fa2ca9c87a0/build01/hydra-declarative-projects.nix#L5 So the only other alternative I see is to have a jobset on the official hydra. There might be also aarch64 support on travis. I don't know if this is enough to build kernel, but if it is than this is actually the better solution because than everyone can customize their built.